script.js

```diff
@@ -0,0 +1,185 @@
+    this.element.classList.add('modal-dialog');
+    // make dialog focusable in order to catch focus from mouse click or touch
+    // event.
+    this.element.setAttribute('tabIndex', '-1');
+
+    // in order to make vertical align, we need additional container
+    this.outerContainer = document.createElement('div');
+    this.outerContainer.classList.add('outer-container');
+
+    this.container = document.createElement('div');
+    this.container.classList.add('container');
+
+    this.messageElement = document.createElement('div');
+    this.messageElement.classList.add('modal-dialog-message');
+
+    this.buttonGroup = document.createElement('div');
+    this.buttonGroup.classList.add('modal-dialog-button-group');
+    this.buttonGroup.setAttribute('smart-bubbles', 'true');
+
+    this.container.appendChild(this.messageElement);
+    this.container.appendChild(this.buttonGroup);
+
+    this.outerContainer.appendChild(this.container);
+    this.element.appendChild(this.outerContainer);
+    document.body.appendChild(this.element);
+
+    this.element.addEventListener('opened', this);
+    this.element.addEventListener('closed', this);
+    this.smartBubble.addEventListener('all-items-bubbled', this);
+  }
+
+  var proto = {};
+
+  proto.open = function(message, buttonSettings, onCancel) {
+    this.isOpened = true;
+    message = message || '';
+
+    // onCancel is triggered when ESC key is pressed.
+    this.onCancel = onCancel;
+    this.buttonElements = [];
+    this.buttonSettings = buttonSettings;
+    this.messageElement.textContent = message;
+    this.messageElement.setAttribute('data-l10n-id', message);
+    this.buttonGroup.innerHTML = '';
+
+    this.messageElement.classList.remove('hidden');
+    if (message.length === 0) {
+      this.messageElement.classList.add('hidden');
+    }
+
+    // Set up every button
+    buttonSettings.forEach(function(buttonSetting, index) {
+      var button = document.createElement('smart-button');
+      button.setAttribute('type', 'circle-text');
+      button.setAttribute('data-l10n-id', buttonSetting.text || 'OK');
+      button.textContent = buttonSetting.text || 'OK';
+      button.classList.add(buttonSetting.class || 'confirm');
+      button.addEventListener('click', function() {
+        // Click action will be handled in closed event
+        this._clickedIndex = index;
+        this.element.close();
+      }.bind(this));
+      this.buttonElements.push(button);
+      this.buttonGroup.appendChild(button);
+    }.bind(this));
+
+    this.element.classList.add('visible');
+    this.element.open();
+  };
+
+  proto.startKeyNavigation = function() {
+    this.element.addEventListener('keydown', this);
+    this._focusedIndex = -1;
+    if (this.buttonElements.length > 0) {
+      this._focusedIndex = 0;
+      this.focus();
+    }
+  };
+
+  proto.focus = function() {
+    var elem = this.buttonElements[this._focusedIndex];
+    if (elem.focus && (typeof elem.focus) === 'function') {
```

Comment:
If `this._focusedIndex` is -1, you may get an exception. Please check `elem` before use it.